### PR TITLE
Adding ssh test to allow check for CPU usage

### PR DESCRIPTION
**Pull request by thaneunsoo**
Adding ssh test package to run tests that require ssh to run. Also adjusted downloadsshkeys.go to send a byte instead of string to be used in the node object.

## Issue: 

https://github.com/rancher/qa-tasks/issues/451
 
## Problem

There were tickets around CPU usage being too high and we wanted to add automated checks to more consistently check CPU usage during releases.
 
## Solution

Adding this automation will allow us to more frequently check if the CPU Usage issue were to ever come back. Also with downloadsshkeys.go we are now able to run any ssh related tests within the provisioning package. Currently just added to rke2 package, but since Node object is also used in rke1 and k3s, I will also add there after approval.
 
### Manual Testing

Tested the actual ssh command manually multiple times to ensure that it works.

### Automated Testing

Will run jenkins job

## QA Testing Considerations

The CPU check will be stored under the ssh package, and with future additions to that package users can just call the function from the ssh package.
 
### Regressions Considerations

Currently there are no regressions as the ssh key being changed from string to byte is allowing the node object to use the ssh key, which before it wasn't allowed. No other objects are being changed.

### CODE REVIEW

File: tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/provisioning_node_driver_test.go:258]: Nit; can you remove this comment?

File: tests/v2/validation/provisioning/ssh.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/ssh.go:3]: Nit; I think the comment here can be useful to have as a broad overview of what this file is. As such, maybe we can expand it a bit on this to be more specific as opposed to executable tests?

File: tests/v2/validation/provisioning/ssh.go
Comment (markusewalker) [reply tests/v2/validation/provisioning/ssh.go:21]: I would say you could just define a `const` block in this `ssh.go` file and just reference that way. Since additional tests are going to be made, it's possible more hard-code values will come, so we can just have that `const` block now.

File: tests/v2/validation/provisioning/ssh.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/ssh.go:11]: You can remove this comment since you already have the defined `const`.

File: tests/v2/validation/provisioning/ssh.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/ssh.go:3]: Maybe @igomez06 can speak more to this...does it matter if the comment is one line like this or do we want to break this up to another line like we do with other functions in the framework?

Either way, I like that it's more descriptive than before!

File: tests/v2/validation/provisioning/ssh.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/ssh.go:19]: Same as above, you can remove this comment. The comment for the function gives the user an explanation as to what is going on here.

File: tests/v2/validation/provisioning/ssh.go
Comment (thaneunsoo) [reply tests/v2/validation/provisioning/ssh.go]: I'll just split into multiple lines to keep it consistent with other comments

**Review by markusewalker — COMMENTED** [review]
LGTM, once we get a Jenkins run, I'll approve.

**Review by markusewalker — DISMISSED** [review]
LGTM.

**Review by slickwarren — APPROVED** [review]
lgtm

**Review by markusewalker — APPROVED** [review]

**Review by igomez06 — APPROVED** [review]
Looks good

**Pull request merged**
